bnelson777/id.ly
# IL 130: implement log in/ register support in app

-Reformatted Login and Register components to match style of the rest of the app
-Implemented usage of bcrypt and rnsi for login encryption
-Implemented a listener that prompts user to login when app is put in the background
-Removed Login and Register components from home screen

Testing steps
1. Delete, npm install and rebuild app
2. Ensure that the first time app is opened you are prompted to set a password. 
  -Attempt to enter 2 passwords that do not match, ensure that the app does not allow you to move 
    forward
  -Enter two matching passwords and ensure the app permits you to the home screen
     **Important: Do not forget the password**
3. Minimize the app. Ensure the app prompts you to re-login when re-opened.
    (Minimizing the app is how the user is logged out)

**Note: Deep linking functionality currently not working with this ticket. We can either test and submit as is and do the deep link portion later, or complete the change by adding deep linking.**
